### PR TITLE
Camera to actuator associations UI

### PR DESCRIPTION
**Pull request by stefanodellosa-personal**
Introduce new dialog to allow associations between available cameras and actuators

### CODE REVIEW

File: src/domain/ftp_camera.py [hunk 1]
```diff
@@ -10,21 +11,30 @@
...
+            "actuators": actuators,
         }
 
     @staticmethod
     def deserialize(data):
         """Method to deserialize FTP Camera object from file."""
-        return FTPCamera(data["id"], data["ftp_folder"], data["enabled"])
+        actuators = []
+        for key in data["actuators"]:
+            actuators.append(Actuator.actuators[key])
```
Comment (cdimauro) [inline src/domain/ftp_camera.py:38]: actuators = [Actuator.actuators[key] for key in data["actuators"]]

**Review by cdimauro — APPROVED** [review]

File: src/domain/ftp_camera.py [hunk 2]
```diff
@@ -10,21 +11,28 @@
...
             "ftp_folder": self.ftp_folder,
+            "actuators": actuators,
         }
 
     @staticmethod
     def deserialize(data):
         """Method to deserialize FTP Camera object from file."""
-        return FTPCamera(data["id"], data["ftp_folder"], data["enabled"])
+
+        actuators = [Actuator.actuators[key] for key in data["actuators"]]
```
Comment (hexfati) [inline src/domain/ftp_camera.py:37]: Check if "actuators" in data to avoid KeyError

Comment (cdimauro) [reply src/domain/ftp_camera.py:37]: actuators = [Actuator.actuators[key] for key in data["actuators"]] if "actuators" in data else []

File: src/domain/configure_actuators.py [hunk 3]
```diff
@@ -41,6 +41,7 @@ def __init__(self):
         self.removed_actuators = []
         self.actuator_server = None
         self.actuator_server_thread = None
+        self.removed_rows = []
```
Comment (cdimauro) [inline src/domain/configure_actuators.py:44]: self.removed_rows = set()

File: src/domain/configure_actuators.py [hunk 4]
```diff
@@ -196,12 +200,13 @@ def remove_actuator(self):
                 actuator_id_ln = self.findChild(QLineEdit, f"lineEdit_actuator_id_{i}")
                 if radiobtn.isChecked() and actuator_id_ln:
                     self.removed_actuators.append(actuator_id_ln.text())
+                    # QGridLayout does not remove the row, only widgets so we store deleted rows
+                    self.removed_rows.append(i)
```
Comment (cdimauro) [inline src/domain/configure_actuators.py:204]: self.removed_rows.add(i)

**Review by cdimauro — APPROVED** [review]

File: src/domain/actuators_management_dialog.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+
+class DialogCameraActuatorManagement(QDialog):
+    """Dialog to edit Actuator(s) list of a given camera"""
+
+    def __init__(self, camera, parent=None):
+        super().__init__(parent)
+        self.setWindowTitle(f"Manage Actuators for Camera ID: {camera.id}")
+        self.camera = camera
+        self.original_actuators = list(camera.actuators)  # Save original list for cancel action
```
Comment (cdimauro) [inline src/domain/actuators_management_dialog.py:24]: You can use:
camera.actuators.copy()
instead of:
list(camera.actuators)

**Review by cdimauro — APPROVED** [review]

**Review by cdimauro — APPROVED** [review]

**Pull request merged**